Updated for Koa 2.x -- Removed and Updated Dependencies

douglas-johnson opened the pull request:
# Removed Dependencies 

Replaced three dependencies with more recent Node / JavaScript built in functions.

## co-fs

- The files did not need to be retrieved using generator/yield as far as I could tell.
- fs.readFileSync is passing all the tests and working in my local implementation

## utility

- The version of string.prototype.replace being utilized from this package was there to prevent a regex error when `$&` is present in the markdown.
- Replacing all instances of `$` with `$$` before using them as the replacement value also prevents this problem.
- Also added a test to show this working.

## copy-to

- Now we can use `Object.assign` for this same purpose

# Updated Dependencies

## Supertest

- Got rid of some NPM security warnings by updating to latest Supertest

## markdown-it

- Updated to latest major version

# Updated for Koa 2.x

- Updated to latest Koa
- Fixed error with constructor invocation by adding `new`
- Updated */yield functions to use async/await instead in accordance with: https://github.com/koajs/koa/blob/master/docs/migration.md

# Increased Node Version in Engine

- Async/Await and Object.assign() should both be covered by 8.0.0, thought they may also be available as early as 7.6 if we need to get more fine grained.

douglas-johnson committed:
update package and package lock

douglas-johnson committed:
remove copy-to in favor of object.assign

douglas-johnson committed:
remove utility dependency, upgrade supertest

douglas-johnson committed:
add test to show replace fix is working

douglas-johnson committed:
update to latest Koa invoke with new

douglas-johnson committed:
switch yield to async/await

douglas-johnson committed:
remove co-fs dependency

douglas-johnson committed:
remove some async awaits we don’t need

douglas-johnson committed:
update markdown-it

douglas-johnson committed:
Merge pull request #1 from douglas-johnson/develop

Update for Koa 2.x and Update or Remove Dependencies

douglas-johnson committed:
update node version

douglas-johnson committed:
Merge pull request #2 from douglas-johnson/develop

update node version

douglas-johnson commented:
@dead-horse @fengmk2 I apologize, I forgot to add reviewers to this pull request. Would either of you mind taking a look? Thank you.

fengmk2 commented on index.js:
```diff
@@ -82,24 +82,37 @@ module.exports = function (options) {
-  function* getContent(filepath) {
-    var content = yield fs.readFile(filepath, 'utf8');
+  function getContent(filepath) {
+    var content = fs.readFileSync(filepath, 'utf-8');
```
should keep using the async api of `fs` module.
`readFileSync` will block other requests in the same time.

Please use [mz/fs](https://www.npmjs.com/package/mz) instead of `co-fs`, then use it's promise api.

douglas-johnson replied:
This has been addressed.

fengmk2 commented on package-lock.json:
```diff
@@ -0,0 +1,2941 @@
+{
+  "name": "koa-markdown",
```
ignore the lock file on lib module.

fengmk2 commented:
Only support node >= 8.0.0 in the next major version of koa-markdown.

douglas-johnson committed:
ignore package lock

douglas-johnson committed:
delete package lock

douglas-johnson committed:
use mx/fs

douglas-johnson committed:
Merge pull request #8 from douglas-johnson/revisions-from-main-branch-pr-18

Revision from Pull Request #18 on Original Repo

douglas-johnson commented:
@fengmk2 I updated the master branch on my fork to address those issues.

- Removed and ignored package-lock
- Used mz/fs with async/await syntax instead of readFileSync

mishugana commented:
bump!
